doc/howtos/software-update-server.rst

```diff
@@ -0,0 +1,287 @@
+
+.. note::
+   Each bundle defined generates a non-negligible amount of load, when
+   building images and SW Update repositories.
+
+   It is therefore recommended to keep enabled only those bundles that
+   are effectively useful for the specific use-case targeted.
+
+   In case one wants to modify the set of predefined bundles that come
+   from the defaults of Ostro OS, it is possible to override the variable
+   ``SWUPD_BUNDLES``.
+
+   Doing so will discard anything that was part of the defaults, therefore
+   if any of the pre-defined bundles is still needed, it must be listed
+   again, explicitly.
+
+   Example::
+
+   ``SWUPD_BUNDLES = "sudo_bundle"``
+
+
+
+Step 3: Build the images and the SW Update repositories
+-------------------------------------------------------
+
+The typical invocation for generating the basic Ostro OS image and related
+SW Update stream is::
+
+ $ bitbake ostro-image-swupd
+
+However, if one wants to have pre-installed bundles, then the invocation
+must refer to the specific variant.
+
+  Example (Continuing from Step2)::
+
+    $ bitbake ostro-image-swupd-pre_installed_content
+
+Assuming that the chosen architecture was ``intel-corei7-64``, the yield
+from the command in the Example is:
+
+- The image with the chosen pre-installed bundles::
+
+    tmp-glibc/deploy/images/intel-corei7-64/ostro-image-swupd-pre_installed_content-intel-corei7-64.dsk
+
+- The base image::
+
+    tmp-glibc/deploy/images/intel-corei7-64/ostro-image-swupd-intel-corei7-64.dsk
+
+- The work directory for generating SW Update repository::
+
+    tmp-glibc/deploy/swupd/intel-corei7-64/ostro-image-swupd/
+
+  This folder contains both data from intermediate steps and the actual
+  SW Update stream.
+
+  Its location is::
+
+    tmp-glibc/deploy/swupd/intel-corei7-64/ostro-image-swupd/www/
+
+  This folder will contain also data related to following builds and it is
+  the one that must be exposed to the device in need of maintenance, through
+  a web server, for example nginx or apache.
+
+
+Step 4: Create an update from the previous step
+-----------------------------------------------
+
+Continuing with the previous example, one possible enhancement is to add a
+new bundle, but it would have been also possible to modify the content of
+the existing bundle(s).
+
+To keep the execution simple, the bundle will contain only one component,
+which is currently missing from the "Version 10" of the distribution: the
+``sed`` command.
+
+The changes required are:
+
+#. Bump the version number::
+
+     OS_VERSION = "20"
+
+#. Add the new bundle to the list of bundles::
+
+     SWUPD_BUNDLES += "sed_bundle"
+
+#. Define the content of the newly created feature::
+
+     BUNDLE_CONTENTS[sed_bundle] = "sed"
+
+#. Had it been required to have the bundle pre-installed, it would have
+   been added to the list of pre-installed bundles::
+
+     SWUPD_IMAGES[pre_installed_content]="\
+       sudo_bundle \
+       sed_bundle \
```

Comment:
this kind of example will lead to confusion since it is so similar to a "single package", which we want to avoid. Let's try to come with something that encompasses multiple packages and is more like a packagegroup or a task, things like: c-devenv, it would pull editors (vim/emacs), gcc, gdb, valgrind, ...